haystack/__init__.py

```diff
@@ -2,33 +2,49 @@
 #
 # SPDX-License-Identifier: Apache-2.0
 
+import sys
+from typing import TYPE_CHECKING
+
+from lazy_imports import LazyImporter
+
+# these imports need to be loaded eagerly
+import haystack.core
 import haystack.logging
 import haystack.tracing
 from haystack.core.component import component
```

Comment:
If we now import `haystack.core` eagerly three lines above, why import haystack.core.component here and then later have it in the else branch with the LazyImporter?
Similarly why do we import haystack.logging here eagerly and then later have it in the else branch with the LazyImporter too?